src/storage/segcache/benches/benchmark.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use core::hash::BuildHasher;
+use core::time::Duration;
+use criterion::{criterion_group, criterion_main, Criterion, Throughput};
+use fasthash::FastHasher;
+use rand::RngCore;
+use rand::SeedableRng;
+use rustcommon_time::*;
+use segcache::SegCache;
+
+pub const MB: usize = 1024 * 1024;
+
+// A very fast PRNG which is appropriate for testing
+pub fn rng() -> impl RngCore {
+    rand_xoshiro::Xoshiro256PlusPlus::seed_from_u64(0)
+}
+
+fn ahash_get_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ahash/get");
+    group.measurement_time(Duration::from_secs(30));
+    group.throughput(Throughput::Elements(1));
+
+    for key_size in [1, 2, 4, 8, 16, 32, 64, 128, 255].iter() {
+        let (keys, _values) = key_values(*key_size, 1_000_000, 0, 0);
+
+        // launch the server
+        let build_hasher = ahash::RandomState::with_seeds(
```

Comment:
What's the relationship between this file and `segcache/example/benchmark.rs` ?

Reply:
I think we can trash the 'example/benchmark' in favor of the one wired up using criterion in 'benches'

Reply:
Trashed the examples/benchmark - the criterion benchmark driven by 'cargo bench' is good enough for quick validation of changes. Anything more formal should be done with rpc-perf/memtier/...